src/gui/newwizard/setupwizardcontroller.cpp

```diff
@@ -166,7 +164,20 @@ void SetupWizardController::changeStateTo(SetupWizardState nextState)
             return;
         }
         case SetupWizardState::CredentialsState: {
-            changeStateTo(SetupWizardState::AccountConfiguredState);
+            // not a fan of performing this job here, should be moved into its own (headless) state IMO
+            auto fetchUserInfoJob = _context->startFetchUserInfoJob();
+
+            connect(fetchUserInfoJob, &CoreJob::finished, this, [this, fetchUserInfoJob] {
+                if (fetchUserInfoJob->success()) {
+                    auto result = fetchUserInfoJob->result().value<FetchUserInfoResult>();
+                    _context->accountBuilder().setDisplayName(result.displayName());
+                    changeStateTo(SetupWizardState::AccountConfiguredState);
+                } else {
+                    _context->window()->showErrorMessage(QStringLiteral("Failed to retrieve user information from server"));
+                    changeStateTo(_currentState->state());
```

Comment:
https://github.com/owncloud/client/blob/7c908d34997423a66d5edd1d2558f79f5e73af71/src/gui/newwizard/setupwizardcontroller.cpp#L111-L114 looks like we would delete us there?

Reply:
This just cleans up the old state after the function has returned. Every possible branch is supposed to create a new state. If you want to, we can make sure this happens by storing a copy and adding another assert next to the others.

Reply:
What if _currentState == nextState

Reply:
ok got it